### ISSUE TITLE
DOI captured incompletely

### ISSUE THREAD

**Issue by bersbersbers** [mdmower/doi-resolver-chrome#5]
For some reason, the DOI here is captured incompletely:
https://www.ncbi.nlm.nih.gov/pubmed/22899574

![image](https://cloud.githubusercontent.com/assets/12128514/21901333/8d26e332-d8f7-11e6-98ca-f60cd4df8a60.png)

**Comment by mdmower** [comment]
That's strange. In the example you've provided, autolinking is working fine for me. Do you have another extension installed on Chrome that could be interfering with DOI Resolver? Also, is this with DOI Resolver 3.1.2 (viewable in Options > About)?

**Comment by bersbersbers** [comment]
I have a host of different extensions installed, but I guess the problem is a different one.

This morning, the problem was gone at first. Then, I repeated the search for this paper with this URL, and it reappeared:
https://www.ncbi.nlm.nih.gov/pubmed/?term=%22Simultaneous+reconstruction+of+activity+and+attenuation+in+time-of-flight+PET%22+2012

With this link, I can reproduce this in a clean profile on stable Chrome with no other extensions installed. I suppose this is related to PubMed trying to highlight the search term "2012", even though it does not actually highlight:
![image](https://cloud.githubusercontent.com/assets/12128514/21922447/f0107644-d96e-11e6-898e-ebf150dc2072.png)

**Comment by mdmower** [comment]
It looks like you've pinpointed the issue. The DOI is split into chunks by their highlighting tag. This could feasibly be worked around by altering my search method to only consider visible text strings, but then I'm nervous about all the hidden tags I could be inadvertently wrapping this way. Thanks for bringing this to my attention, but I'll need to think about how to handle cases like this.

**Comment by bersbersbers** [comment]
If you ever decide to work on this, please consider `<wbr>` tags as well, as inserted by Gmail:

![image](https://user-images.githubusercontent.com/12128514/129533172-d08fb473-b6f1-40d0-b298-e43ed90efa6e.png)

These lead to the same effect apparently.

Related: https://stackoverflow.com/questions/24929372/gmail-inserts-wbr-tags-into-links

**Comment by mdmower** [comment]
I appreciate the report and there's no excuse for the length of time it's taken me to dig into this, but I don't plan to support this feature request. I spent some time re-reviewing the autolink script this week and I'm still content with its safety and speed. The extra logic to look for a DOI in `lorem ipsum 10.<tag>1000</tag>/182 lorem ipsum` would slow the search down considerably. More importantly, though, there is risk in adversely affecting the display of content.

Closing as won't fix, but again, thanks for bringing this to my attention.
